src/roswire/ros1/state.py

```diff
@@ -2,14 +2,54 @@
 __all__ = ("SystemStateProbe",)
 
 import xmlrpc
-from typing import Dict, Sequence, Tuple
+from typing import AbstractSet, Collection, Dict, Mapping, Sequence, Set, Tuple
 
 import attr
 
 from .. import exceptions as exc
 from ..common import SystemState
 
 
+@attr.s(frozen=True, auto_attribs=True, slots=True)
+class ROS1SystemState(SystemState):
+    """Provides a description of the instantaneous state of a ROS system in
```

Comment:
Let's lift the docstring into the `SystemState` interface and omit it from the implementing classes.